sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -17,22 +49,284 @@ class TableCheckpointStore:
     :keyword str api_version:
-            The Storage API version to use for requests. Default value is '2019-07-07'.
-    :keyword str secondary_hostname:
-        The hostname of the secondary endpoint.
+            The Storage API version to use for requests. Default value is '2018-03-28'.
     """
 
-    def __init__(self, **kwargs):
-        pass
+    def __init__(self, table_account_url, table_name, credential=None, **kwargs):
+        # type(str, str, Optional[Any], Any) -> None
+        self._table_client = kwargs.pop("table_client", None)
+        if not self._table_client:
+            api_version = kwargs.pop("api_version", None)
+            if api_version:
+                headers = kwargs.get("headers")
+                if headers:
+                    headers["x-ms-version"] = api_version
+                else:
+                    kwargs["headers"] = {"x-ms-version": api_version}
+            self._table_client = TableClient(
+                table_account_url, table_name, credential=credential, **kwargs
+            )
+            self._cached_table_clients = defaultdict()  # type: Dict[str, TableClient]
+
+    @classmethod
+    def from_connection_string(cls, conn_str, table_name, credential=None, **kwargs):
+         # type: (str, str, Optional[Any], Any) -> TableCheckpointStore
+        """Create TableCheckpointStore from a storage connection string.
+
+        :param str conn_str:
+            A connection string to an Azure Storage account.
+        :param table_name:
+            The table name.
+        :type table_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+            Credentials provided here will take precedence over those in the connection string.
+        :keyword str api_version:
+            The Storage API version to use for requests. Default value is '2018-03-28'.
+        :returns: A table checkpoint store.
+        :rtype: ~azure.eventhub.extensions.checkpointstoretable.TableCheckpointStore
+        """
+        endpoint, credential = parse_connection_str(
+            conn_str=conn_str, credential=None, keyword_args=kwargs
+        )
+        return cls(endpoint, table_name=table_name, credential=credential, **kwargs)
+
+    def __enter__(self):
+        self._table_client.__enter__()
+        return self
+
+    def __exit__(self, *args):
+        self._table_client.__exit__(*args)
+
+    def _get_table_client(self, table_name):
+        result = self._cached_table_clients.get(table_name)
+        if not result:
+            result = self._table_client.get_table_client(table_name)
```

Comment:
I don't think `TableClient` has a method called `get_table_client`. Does it actually?

(Though, I don't see the benefit to having `_get_table_client` instead of just referring to `_table_client` directly.)